lib/plugins/unthreaded-signing/plugin.c

```diff
@@ -125,15 +163,37 @@ sv_signing_plugin_session_setup(const void *private_key, size_t private_key_size
     self = NULL;
   }
 
+  // Preallocate memory for the signatures in the buffer
+  for (int i = 0; i < MAX_BUFFER_LENGTH; i++) {
+    self->out_buffer[i].signature = calloc(1, self->sign_data.max_signature_size);
+    if (!self->out_buffer[i].signature) {
+      // Handle allocation failure
+      sv_signing_plugin_session_teardown((void *)self);
+      return NULL;
+    }
+  }
+
   return self;
 }
 
+static void
+out_buffer_teardown(sv_unthreaded_plugin_t *self)
+{
+  for (int i = 0; i < MAX_BUFFER_LENGTH; i++) {
+    if (self->out_buffer[i].signature) {
+      free(self->out_buffer[i].signature);
+      self->out_buffer[i].signature = NULL;
```

Comment:
This line is not needed since the buffer teardown is done in plugin teardown only which ends all operations.